agent/consul/discoverychain/compile.go

```diff
@@ -874,8 +874,14 @@ RESOLVE_AGAIN:
 
 	targetID := target.ServiceID()
 
-	if err := c.recordServiceProtocol(targetID); err != nil {
-		return nil, err
+	// Only validate protocol if it is not a peered service.
+	// TODO: Add in remote peer protocol validation when building a chain.
+	// This likely would require querying the imported services, which
+	// shouldn't belong in the discovery chain compilation here.
+	if target.Peer == "" {
```

Comment:
Ah so this boils down to "there is not a `service-defaults` for imported services to store the protocol" if I'm understanding.

Reply:
Yup. I was still writing the description up for the ticket, haha. I guess the system automatically assigned you to this PR.